src/codec.cairo

```diff
@@ -1,22 +1,24 @@
 //! Bitcoin binary codec traits, implementations, and helpers.
 
-use super::types::transaction::{Transaction, TxIn, TxOut};
+// use core::fmt::{Display, Formatter, Error};
+use super::types::transaction::{Transaction, TxIn, TxOut, OutPoint};
+use raito::utils::hash::Hash;
 
-pub trait Encode<T> {
+pub trait EncodeTrait<T> {
     /// Convert into bytes and append to the buffer
     fn encode_to(self: T, ref dest: ByteArray);
-
     /// Convert into bytes and return
     fn encode(
-        self: T
+        self: T, _is_segwit: bool,
     ) -> ByteArray {
         let mut dest: ByteArray = Default::default();
         Self::encode_to(self, ref dest);
```

Comment:
You'd need to forward segwit flag to encode_to as well

Reply:
I used _is_segwit to respect the old structure, currently we don't recover marker and flag fields in Transaction. So I did everything without the segwit but I can add it.

Reply:
Sure, let's keep for a follow up